### PR TITLE
added fast-metals option to picca_metal_dmat [minor]

### PR DESCRIPTION
**Pull request by calumgordon**
Calling this option only computes the relevant 4 LYA x Si and CIVxCIV.

### CODE REVIEW

File: bin/picca_metal_dmat.py [hunk 1]
```diff
@@ -416,6 +422,16 @@ def main(cmdargs):
         for index2, abs_igm2 in enumerate(abs_igm_2[index0:]):
             if index1 == 0 and index2 == 0:
                 continue
+
+            if args.fast_metals:
+                default_si_metals = ['SiIII(1207)','SiII(1190)','SiII(1193)','SiII(1260)']
+                if (abs_igm1=='LYA')&(abs_igm2 in default_si_metals):
```
Comment (iprafols) [inline bin/picca_metal_dmat.py:428]: `if abs_igm1 == "Lya" and abs_igm2 in default_si_metals:`

File: bin/picca_metal_dmat.py [hunk 2]
```diff
@@ -416,6 +422,16 @@ def main(cmdargs):
...
+                    pass
+                elif abs_igm1==abs_igm2=='CIV(eff)':
```
Comment (iprafols) [inline bin/picca_metal_dmat.py:430]: `elif abs_igm1 == abs_igm2 and abs_igm1 == "CIV(eff)":`

File: bin/picca_metal_dmat.py (hunk 1 above)
Comment (iprafols) [reply bin/picca_metal_dmat.py:428]: (if kept, see later comments)

File: bin/picca_metal_dmat.py [hunk 3]
```diff
@@ -416,6 +422,16 @@ def main(cmdargs):
...
+                    pass
+                else:
```
Comment (iprafols) [inline bin/picca_metal_dmat.py:432]: why not just put one condition and then the `continue`?  

something like:
```
if not (abs_igm1 == "Lya" and abs_igm2 in default_si_metals) and not (abs_igm1 == "CIV(eff)" and abs_igm1 == abs_igm2):
    continue
```

File: bin/picca_metal_dmat.py (hunk 2 above)
Comment (iprafols) [reply bin/picca_metal_dmat.py:430]: (if kept, see later comments)

File: bin/picca_metal_dmat.py [hunk 4]
```diff
@@ -416,6 +422,16 @@ def main(cmdargs):
         for index2, abs_igm2 in enumerate(abs_igm_2[index0:]):
             if index1 == 0 and index2 == 0:
                 continue
+
+            if args.fast_metals:
+                default_si_metals = ['SiIII(1207)','SiII(1190)','SiII(1193)','SiII(1260)']
```
Comment (iprafols) [inline bin/picca_metal_dmat.py:427]: this should be defined as a constant at the beginning of the file `DEFAULT_SI_METALS= ['SiIII(1207)','SiII(1190)','SiII(1193)','SiII(1260)']`

**Review by iprafols — CHANGES_REQUESTED** [review]

File: bin/picca_metal_dmat.py (hunk 3 above)
Comment (calumgordon) [reply bin/picca_metal_dmat.py]: changed to this

File: bin/picca_metal_dmat.py [hunk 5]
```diff
@@ -313,6 +313,9 @@ def main(cmdargs):
     cf.alpha_abs[args.lambda_abs] = cf.alpha
     for metal in args.abs_igm:
         cf.alpha_abs[metal] = args.metal_alpha
+    
+    #store the default silicon metals modelled in the CF/XCF
+    DEFAULT_SI_METALS = ['SiIII(1207)','SiII(1190)','SiII(1193)','SiII(1260)']
```
Comment (iprafols) [inline bin/picca_metal_dmat.py:318]: This should be placed at the top of the file, in line 20

**Review by iprafols — CHANGES_REQUESTED** [review]

**Comment by iprafols** [comment]
also note that the tests are failing now

**Review by iprafols — APPROVED** [review]

**Pull request merged**